groundcontrol.kv

```diff
@@ -280,9 +280,11 @@
     Button:
         text: 'Open File'
         on_press: root.openFile()
+		disabled: app.data.uploadFlag == 1
```

Comment:
I didn't you could reference data like this...why are we not doing this everywhere?...beautiful!

Reply:
I'm beginning to like Kivy... but I still get confused about self.x vs root.x vs app.x vs x
I tried root.data, but "root" is the class, and at construct time, it doesn't have "data" (that gets done just a bit later.

And I'm not sure you get to format it.  Text: "0.2f"%root.data.ZPos explodes.

Reply:
I can't tell you how much time I've spent struggling with things not having access to data at construct time. I love kivy, but that was one thing that really bugged me...looks like I just missed the option to call app.data which would have made it easy.

Does it work from within modules too, or just the .kv file?

Reply:
I've only done it in the .kv file so far... that was slicker than expected....

Reply:
On a completely different thread, can you tell me how saving data is supposed to work?  I think Issue #591 is close, but the load/save just isn't.